File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

// dfs 돌렸는데 시간초과나서 그리디
string solution(string number, int k) {
    string answer = "";
    
    for(int i=0; i<number.length(); i++){
        while(k>0 && answer != "" && number[i] > answer.back()){
            answer.pop_back();
            k--;
        }
        
        if(answer.length() < number.length()-k)
            answer += number[i];
    }
    
    return answer;
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    let answer = '';
    
    let tmp = [];
    for(let i=0; i<number.length; i++){
        while(k>0 && number[i] > tmp[tmp.length-1]){
            tmp.pop();
            k--;
        }
        
        if(tmp.length < number.length-k)
            tmp.push(number[i]);
    }
    
    answer = tmp.join('');
    
    return answer;
}